algorithms/CPlusPlus/Recursion/string-subsequences.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+Description: Recursion Program to find subsequences of a string.
+
+Example:
+Original String = "abc"
+Output = {"", "c" , "b", "bc", "a", "ac", "ab", "abc"}
+
+Appraoch:
+Subsequences are the every possible sequences that can be formed from a string
+from a empty string to every character in the string.
+
+In the below code we return the array size which contains our subsequences.
+In every recursion call the function get the double subsequences of what it was
+before by adding single character to the previous call.
+*/
+
+#include <iostream>
```

Comment:
```suggestion
#include <iostream>
#include <string>
```

Make sure to include all the required headers, even though some implementations include string header which implementing parts of iostream, but they are not obligated to do so.